aiida_aurora/workflows/cycling_sequence.py

```diff
@@ -0,0 +1,130 @@
+            continue
+
+        reference_keys.remove(namekey)
+
+    for remaining_key in reference_keys:
+        error_message += f'protocol {remaining_key} has no settings\n'
+
+    if len(error_message) > 0:
+        return error_message
+
+
+class CyclingSequenceWorkChain(WorkChain):
+    """This workflow represents a process containing a variable number of steps."""
+
+    @classmethod
+    def define(cls, spec):
+        """Define the process specification."""
+        # yapf: disable
+        super().define(spec)
+        spec.input(
+            "battery_sample",
+            valid_type=BatterySampleData,
+            help="Battery sample to be used."
+        )
+        spec.input(
+            "tomato_code",
+            valid_type=orm.Code,
+            help="Tomato code to use."
+        )
+
+        spec.input_namespace(
+            "protocols",
+            dynamic=True,
+            valid_type=CyclingSpecsData,
+            help="List of experiment specifications."
+        )
+        spec.input_namespace(
+            "control_settings",
+            dynamic=True,
+            valid_type=TomatoSettingsData,
+            help="List of experiment control settings."
+        )
+        spec.output_namespace(
+            "results",
+            dynamic=True,
+            valid_type=orm.ArrayData,
+            help="Results of each step by key."
+        )
+
+        spec.inputs.validator = validate_inputs
+
+        spec.outline(
+            cls.setup_workload,
+            while_(cls.has_steps_remaining)(
+                cls.run_cycling_step,
+                cls.inspect_cycling_step,
+            ),
+            cls.gather_results,
+        )
+
+        spec.exit_code(
+            401,
+            'ERROR_IN_CYCLING_STEP',
+            message='One of the steps of CyclingSequenceWorkChain failed'
+        )
+
+    def setup_workload(self):
+        """Takes the inputs and wraps them together."""
+        self.worksteps_keynames = list(self.inputs['protocols'].keys())
+
+    def has_steps_remaining(self):
+        """Check if there is any remaining step."""
+        return len(self.worksteps_keynames) > 0
+
+    def inspect_cycling_step(self):
+        """Verify that the last cycling step finished successfully."""
+        last_subprocess = self.ctx.subprocesses[-1]
+
+        if not last_subprocess.is_finished_ok:
+            pkid = last_subprocess.pk
+            stat = last_subprocess.exit_status
+            self.report(f'Cycling substep <pk={pkid}> failed with exit status {stat}')
+            return self.exit_codes.ERROR_IN_CYCLING_STEP
+
+    def run_cycling_step(self):
+        """Run the next cycling step."""
+        current_keyname = self.worksteps_keynames.pop(0)
+        inputs = {
+            'code': self.inputs.tomato_code,
+            'battery_sample': self.inputs.battery_sample,
+            'technique': self.inputs.protocols[current_keyname],
```

Comment:
`'technique'` should be `'protocol'`. As discussed, changes need to be made in `cycler.py` to uniform the language. As a reminder, `protocols` are a set of ``protocol` instances, e.g. "formation", "cycling", each made up of a set of `technique` instances, e.g. "OCV", "CC", "LOOP", etc.